master/buildbot/process/buildstep.py

```diff
@@ -812,7 +813,8 @@ def addLogWithFailure(self, why, logprefix=""):
         # helper for showing exceptions to the users
         try:
             yield self.addCompleteLog(logprefix + "err.text", why.getTraceback())
-            yield self.addHTMLLog(logprefix + "err.html", formatFailure(why))
+            yield self.addHTMLLog(logprefix + "err.html",
+                                  bytes2NativeString(formatFailure(why)))
```

Comment:
I would expect addHTMLLog to do the ascii2unicode by itself (will probably save you a lot of fixes)